pkg/convert/pprof/streaming/vt_sample.go

```diff
@@ -0,0 +1,293 @@
+package streaming
+
+import (
+	"fmt"
+
+	"io"
+)
+
+func (p *VTStreamingParser) parseSampleVT(buffer []byte) error {
+	p.tmpSample.resetSample()
+	err := p.tmpSample.UnmarshalSampleVT(buffer, &p.tmpLabel)
+	if err != nil {
+		return err
+	}
+
+	for i := range p.tmpSample.tmpStackLoc {
+		err = p.addStackLocation(p.tmpSample.tmpStackLoc[i])
+		if err != nil {
+			return err
+		}
+	}
+	reverseStack(p.tmpSample.tmpStack)
+
+	p.createTrees(p.newCache)
+
+	return nil
+}
+
+// revive:disable-next-line:cognitive-complexity,cyclomatic necessary complexity
+func (s *sample) UnmarshalSampleVT(dAtA []byte, tmpLabel *label) error {
```

Comment:
It does not seem we modify it here (we should not at least as far as I can see) so please consider passing it by value - this is way cheaper to pass label by value

Reply:
I mean, we do not need a tmpLabel at all: it is better to allocate small types like below on the stack in the function body: 

```go

 type line struct {
 	functionID uint64
 }

 type label struct {
 	k, v int64
 }

```